### PR TITLE
Fix Node JS http redirect

### PR DESCRIPTION
**Pull request by mdittmer**
for cases such as 'http://example.com/frobinator' redirect to '/new-frobinator' (notice no protocol or host in new URL)

### CODE REVIEW

File: src/lib/node/net.js [hunk 1]
```diff
@@ -656,16 +656,36 @@ foam.CLASS({
       class: 'Boolean',
       name: 'followRedirect',
       value: true
+    },
+    {
+      name: 'urlLib',
+      value: require('url')
```
Comment (adamvy) [inline src/lib/node/net.js]: This isn't serializable.  I would just keep doing require('url') where needed unless it needs to be saved for performance reasons.

File: src/lib/node/net.js [hunk 2]
```diff
@@ -656,16 +656,36 @@ foam.CLASS({
...
+      if ( data.protocol ) this.protocol = data.protocol.slice(0, -1);
+      if ( data.hostname ) this.hostname = data.hostname;
       if ( data.port ) this.port = data.port;
-      this.path = data.path;
+      if ( data.path ) this.path = data.path;
+
+      return this;
+    },
+
+    function copyUrlFrom(other) {
```
Comment (adamvy) [inline src/lib/node/net.js:677]: Why does the standard copyFrom semantics not work for this?  Can we fix the model so that it does?

Comment (mdittmer) [reply src/lib/node/net.js:677]: Well, for one thing, we don't want to copy everything, only the URL-related stuff. So using copyFrom would be something like

```js
foo.copyFrom({
  protocol: bar.protocol,
  /* ... */
})
```

For another thing, port is copied conditionally, which further mucks with a `copyFrom()` call. That's why I wanted to consolidate the careful-imperative-copy in to a helper method.

File: src/lib/node/net.js (hunk 1 above)
Comment (kgrgreer) [reply src/lib/node/net.js]: Can you make it a factory:?

**Pull request merged**